static/js/publisher/market/market.js

```diff
@@ -197,6 +197,12 @@ function initForm(config, initialState, errors) {
 
       // make sure we don't warn user about leaving the page when submitting
       ignoreChangesOnUnload = true;
+
+      // disable button and show spinner when loading is long
+      disableSubmit();
+      setTimeout(() => {
+        submitButton.classList.add('has-spinner');
+      }, 2000);
```

Comment:
2 seconds quick enough? I would maybe reduce it to a second? If not less?

Reply:
@Lukewh 2 sec is as per @matthewpaulthomas suggestion (https://github.com/canonical-websites/snapcraft.io/issues/602#issuecomment-389116739), also we use similar delay in BSI

Reply:
Ignore me then :)!